std/zip.d

```diff
@@ -810,141 +885,60 @@ public:
+        ulong next;
 
         if (isZip64)
         {
-            // Read Zip64 record data
-            ulong eocdOffset = getUlong(i + 8);
-            enforce!ZipException(eocdOffset + zip64EndOfCentralDirLength <= _data.length,
-                                 "corrupted directory");
-
-            i = to!uint(eocdOffset);
-            enforce!ZipException(_data[i .. i + 4] == zip64EndOfCentralDirSignature,
+            auto z64eocdlr = getRecord!Zip64EndOfCentralDirLocatorRecord(eocdrOffset
+                                                                         - Zip64EndOfCentralDirLocatorRecord.sizeof);
+            auto z64eocdrOffset = fromLE!ulong(z64eocdlr.offset);
+            auto z64eocdr = getRecord!Zip64EndOfCentralDirRecord(z64eocdrOffset);
+
+            count = fromLE!ulong(z64eocdr.entries);
+            next = fromLE!ulong(z64eocdr.offset);
+            auto size = fromLE!ulong(z64eocdr.size);
+            end = next + size;
+            auto entriesOnThisDisk = fromLE!ulong(z64eocdr.entriesOnThisDisk);
+
+            enforce!ZipException(z64eocdr.signature == zip64EndOfCentralDirSignature,
                                  "invalid Zip EOCD64 signature");
-
-            ulong eocd64Size = getUlong(i + 4);
-            enforce!ZipException(eocd64Size + i - 12 <= data.length,
-                                 "invalid Zip EOCD64 size");
-
-            // zip64 end of central dir record
-            removeSegment(i, cast(uint) (i + 12 + eocd64Size));
-
-            ulong numEntriesUlong = getUlong(i + 24);
-            ulong totalEntriesUlong = getUlong(i + 32);
-            ulong directorySizeUlong = getUlong(i + 40);
-            ulong directoryOffsetUlong = getUlong(i + 48);
-
-            enforce!ZipException(numEntriesUlong <= uint.max,
+            enforce!ZipException(entriesOnThisDisk <= uint.max,
                                  "supposedly more than 4294967296 files in archive");
-
-            enforce!ZipException(numEntriesUlong == totalEntriesUlong,
+            enforce!ZipException(entriesOnThisDisk == count,
                                  "multiple disk zips not supported");
-
-            enforce!ZipException(directorySizeUlong <= i && directoryOffsetUlong <= i
-                                 && directorySizeUlong + directoryOffsetUlong <= i,
+            enforce!ZipException(size <= z64eocdrOffset && next <= z64eocdrOffset && end <= z64eocdrOffset,
                                  "corrupted directory");
 
-            directoryCount = to!uint(totalEntriesUlong);
-            directorySize = to!uint(directorySizeUlong);
-            directoryOffset = to!uint(directoryOffsetUlong);
+            // The extensible data segment is not yet used, but we need to query it to make sure
+            // it doesn't overlap anything else. Might be provided to the user in the future.
+            auto extensibleData = getSegment(fromLE!ulong(z64eocdlr.offset) + Zip64EndOfCentralDirRecord.sizeof,
```

Comment:
this doesn't seem used.

Reply:
Yes, that's why I wrote the comment above - the `getSegment` needs to be called to check for invalid input. @CyberShadow once suggested to make this data available to the user, but I'd like to address this in a separate PR, as it will introduce a new public symbol.